server/src/main/java/org/elasticsearch/index/mapper/BooleanFieldMapper.java

```diff
@@ -409,7 +420,15 @@ protected void parseCreateField(DocumentParserContext context) throws IOExceptio
                 value = nullValue;
             }
         } else {
-            value = context.parser().booleanValue();
+            try {
+                value = context.parser().booleanValue();
+            } catch (IllegalArgumentException e) {
+                if (ignoreMalformed.value() && context.parser().currentToken().isValue()) {
+                    context.addIgnoredField(mappedFieldType.name());
```

Comment:
The NumberFieldMapper has some additional code here that conditionally stores the field for retrieval via the new "synthetic source" feature:
```
     if (storeMalformedFields) {
                    // Save a copy of the field so synthetic source can load it
                    context.doc().add(IgnoreMalformedStoredValues.storedField(name(), context.parser()));
                }
```
I don't know if we want this for the boolean type as well, let me check this with someone with more context in that area and get back to you.

Reply:
I need to know more about synthetic source.

Reply:
Thanks for resolving the other comments. How to proceed with this?